src/Symfony/Component/Messenger/RoutableMessageBus.php

```diff
@@ -45,14 +45,15 @@ public function dispatch($envelope, array $stamps = []): Envelope
 
         /** @var BusNameStamp $busNameStamp */
         $busNameStamp = $envelope->last(BusNameStamp::class);
-        if (null === $busNameStamp) {
-            throw new InvalidArgumentException('Envelope does not contain a BusNameStamp.');
+        $busNameStampName = MessageBusInterface::class;
+        if (null !== $busNameStamp) {
+            $busNameStampName = $busNameStamp->getBusName();
         }
 
-        if (!$this->busLocator->has($busNameStamp->getBusName())) {
-            throw new InvalidArgumentException(sprintf('Invalid bus name "%s" on BusNameStamp.', $busNameStamp->getBusName()));
+        if (!$this->busLocator->has($busNameStampName)) {
+            throw new InvalidArgumentException(sprintf('Invalid bus name "%s" on BusNameStamp.', $busNameStampName));
```

Comment:
You need to change this exception to be coherent with the new behaviour:
```suggestion
            throw new InvalidArgumentException(sprintf('Bus name "%s" does not exist.', $busName));
```